src/Microsoft.ComponentDetection.Common/DependencyGraph/ComponentRecorder.cs

```diff
@@ -173,6 +204,19 @@ public DetectedComponent GetComponent(string componentId)
             }
         }
 
+        public void RegisterPackageParseFailure(string skippedComponent)
+        {
+            if (skippedComponent == null)
+            {
+                throw new ArgumentNullException(paramName: nameof(skippedComponent));
+            }
+
+            lock (this.registerSkippedLock)
```

Comment:
This lock adds unnecessary overhead, the concurrent data structures are already thread-safe (and often times lock-free!)

Reply:
Done.